### DISCUSSION TITLE
Add tenants in version 0.3.5

### DISCUSSION THREAD

**Discussion by pmarchal25** [mattieserver/netbox-topology-views#327] [General]
Hi, can i add tenants to the filter of the netbox topology for the version 0.3.5? I

**Comment by mattieserver** [comment]
Both tenant & tenant group are supported in the search form.

**Reply by pmarchal25** [reply]
Hello matiieserver,

how can i add both to the filter? I have tried to add them in the views.py file in \netbox-topology-views-0.5.3\netbox_topology_views\api\views.py and the information in the html file to create it, but i can't see the filter, how could i do it?

Thank you so much.

Pedro

**Reply by mattieserver** [reply]
I am not sure what you mean?

The ui shows the filter:
![image](https://github.com/mattieserver/netbox-topology-views/assets/3049868/dcc64f5a-544a-410e-82cf-00444c976d15)

**Reply by pmarchal25** [reply]
Hello mattiserver, I have installed the version 0.3.5 because i have netbox version 2.11.11 but i think that this version of the plugin doesn't support tenants. I only can filter these ones:
<img width="224" alt="Captura" src="https://github.com/mattieserver/netbox-topology-views/assets/124051970/7347a9c9-0d34-4792-a209-1a2f85eba29a">
I would like that in my search field appear the tenant group and the tenant, how could i do it?

Thanks.

Pedro

**Reply by mattieserver** [reply]
Ah, sorry I didn't notice the version you posted.

I would try to update your netbox, so much has changed in the plugin and the plugin framework for netbox that it is hard to say what is needed.

If you were to try it anyway you will have to change the javascript and the template aswel since in that version the forms where not used yet.

**Reply by pmarchal25** [reply]
Hi matieserver, the problem is that we can't use netbox version 3, because we want the secrets in our netbox (and not with an external credential management like the version 3 of netbox use). Yes i have to change the index.htm in order to the tenants filter appears and the javascript (i did both changes but it doesn't work... i think that maybe i need to add the models of tenant?)

Thanks.
Pedro

**Reply by dreng** [reply]
> the problem is that we can't use netbox version 3, because...

Isn't that always the problem with any other software? "We'd like to use Excel 2000...", "We need to use Windows XP...", you name it. Honestly, you don't! Change your mindset, change your workflow. It is rarely a good idea to hold on to outdated software forever and your version is very outdated.

**Reply by dreng** [reply]
Have you tried this plugin? https://github.com/Onemind-Services-LLC/netbox-secrets

**Reply by pmarchal25** [reply]
Hi Dreng, I know what you said, but the problem is that i am not te boss in the project... I told to my boss that the best way should be install the last version of Netbox and migrate the Database to this new version, but they don't want to do this release...Then i have to find a solution with the version that i have..

Thank you so much, i am trying to modify the code with the aim to use the tenants filter in the 2.11.11 version of netbox.

**Comment by pmarchal25** [comment]
Hi,

In the index.html i have added this:
<img width="497" alt="tenants" src="https://github.com/mattieserver/netbox-topology-views/assets/124051970/ef8e60d6-00d1-40d9-9653-3df672649e4f">


and in \netbox-topology-views-0.5.3\netbox_topology_views\api/views.py:

In def _filter(self, site, role, name, tags, region, location, tenant):
        if tenant is not None:
            filter_devices= filter_devices.filter(Tenant__id__in=tenant)**
        return filter_devices
<img width="566" alt="tenantsViews" src="https://github.com/mattieserver/netbox-topology-views/assets/124051970/5aee9668-2bff-4fd6-934e-44913a3598a3">
In search method: 

<img width="533" alt="tenantsViews2" src="https://github.com/mattieserver/netbox-topology-views/assets/124051970/1832a9aa-bbd5-4895-864b-b87b184c6f94">

        tenants = request.query_params.getlist('tenants[]',None)
        if tenants == []:
            tenants = None
        devices = self._filter(sites, devicerole, name, tags, regions, locations,tenants)


I think that maybe i must create the model tenant with the aim to use it on the code? I am not sure, thanks for your help
